src/util/integer_gmp_imp.cpp

```diff
@@ -402,28 +430,61 @@ unsigned int Integer::getUnsignedInt() const
   return (unsigned int)d_value.get_ui();
 }
 
-bool Integer::fitsSignedLong() const { return d_value.fits_slong_p(); }
-
-bool Integer::fitsUnsignedLong() const { return d_value.fits_ulong_p(); }
-
 long Integer::getLong() const
 {
-  long si = d_value.get_si();
-  // ensure there wasn't overflow
-  CheckArgument(mpz_cmp_si(d_value.get_mpz_t(), si) == 0,
+  // ensure there it fits
+  CheckArgument(mpz_fits_slong_p(d_value.get_mpz_t()) != 0,
                 this,
                 "Overflow detected in Integer::getLong().");
-  return si;
+  return d_value.get_si();
 }
 
 unsigned long Integer::getUnsignedLong() const
 {
-  unsigned long ui = d_value.get_ui();
-  // ensure there wasn't overflow
-  CheckArgument(mpz_cmp_ui(d_value.get_mpz_t(), ui) == 0,
+  // ensure that it fits
+  CheckArgument(mpz_fits_ulong_p(d_value.get_mpz_t()) != 0,
                 this,
                 "Overflow detected in Integer::getUnsignedLong().");
-  return ui;
+  return d_value.get_ui();
+}
+
+int64_t Integer::getSigned64() const
+{
+  if constexpr (sizeof(int64_t) == sizeof(signed long int))
+  {
+    return getLong();
+  }
+  else
+  {
+    if (mpz_fits_slong_p(d_value.get_mpz_t()) != 0)
+    {
+      return getLong();
+    }
+    // ensure there isn't overflow
+    CheckArgument(mpz_sizeinbase(d_value.get_mpz_t(), 2) < 64,
```

Comment:
Add a comment here why it is < 64 in the signed case (since `mpz_sizeinbase` ignores the sign bit).